auth/lib/store/code_store.dart

```diff
@@ -22,28 +23,51 @@ class CodeStore {
     _authenticatorService = AuthenticatorService.instance;
   }
 
-  Future<List<Code>> getAllCodes({AccountMode? accountMode}) async {
+  Future<AllCodes> getAllCodes({AccountMode? accountMode}) async {
     final mode = accountMode ?? _authenticatorService.getAccountMode();
     final List<EntityResult> entities =
         await _authenticatorService.getEntities(mode);
     final List<Code> codes = [];
+    bool hasError = false;
+
     for (final entity in entities) {
-      final decodeJson = jsonDecode(entity.rawData);
-      final code = Code.fromRawData(decodeJson);
-      code.generatedID = entity.generatedID;
-      code.hasSynced = entity.hasSynced;
-      codes.add(code);
+      try {
+        final decodeJson = jsonDecode(entity.rawData);
+
+        late Code code;
+        if (decodeJson is String && decodeJson.startsWith('otpauth://')) {
+          code = Code.fromOTPAuthUrl(decodeJson);
+        } else {
+          code = Code.fromExportJson(decodeJson);
+        }
+        code.generatedID = entity.generatedID;
+        code.hasSynced = entity.hasSynced;
+        codes.add(code);
+      } catch (e) {
+        hasError = true;
+        _logger.severe("Could not parse code", e);
+      }
     }
 
     // sort codes by issuer,account
-    codes.sort((a, b) {
-      final issuerComparison = compareAsciiLowerCaseNatural(a.issuer, b.issuer);
+    codes.sort((firstCode, secondCode) {
+      if (secondCode.isPinned && !firstCode.isPinned) return 1;
+      if (!secondCode.isPinned && firstCode.isPinned) return -1;
+
+      final issuerComparison =
+          compareAsciiLowerCaseNatural(firstCode.issuer, secondCode.issuer);
       if (issuerComparison != 0) {
         return issuerComparison;
       }
-      return compareAsciiLowerCaseNatural(a.account, b.account);
+      return compareAsciiLowerCaseNatural(
+        firstCode.account,
+        secondCode.account,
+      );
     });
-    return codes;
+    return AllCodes(
+      codes: codes,
+      state: hasError ? AllCodesState.error : AllCodesState.value,
```

Comment:
This error state should be part of Code? Otherwise if any code parsing fails, we will end up showing error state for all valid codes too.
```dart
Code {
 Error? err;
 bool get hasError =? err != nil;
  Code withError(Error err) {
  }
```
}

Reply:
I have kept both, store error and also store state of allcodes to keep it handy while checking if any error was there while fetching codes.